ebi_eva_common_pyutils/spring_properties.py

```diff
@@ -26,90 +28,160 @@ def __init__(self, maven_profile, private_settings_file):
+            self._common_properties(assembly_accession=target_assembly, read_preference='secondaryPreferred',
+                                    chunk_size=chunk_size),
+            self._accessioning_properties(instance=instance),
+            {
+                'spring.batch.job.names': 'CREATE_SUBSNP_ACCESSION_JOB',
+                'parameters.assemblyReportUrl': self._format_str('file:{0}', assembly_report),
+                'parameters.contigNaming': 'NO_REPLACEMENT',
+                'parameters.fasta': fasta,
+                'parameters.forceRestart': 'false',
+                'parameters.projectAccession': project_accession,
+                'parameters.taxonomyAccession': taxonomy_accession,
+                'parameters.vcfAggregation': aggregation,
+                'parameters.vcf': vcf_file,
+                'parameters.outputVcf': output_vcf
+            },
+        )
 
     def get_remapping_extraction_properties(self, *, taxonomy='', source_assembly='', fasta='', assembly_report='',
                                             projects='', output_folder='.'):
         """Properties for remapping extraction pipeline."""
-        return self._common_properties(source_assembly) + f'''
-spring.batch.job.names=EXPORT_SUBMITTED_VARIANTS_JOB
-parameters.taxonomy={taxonomy}
-parameters.fasta={fasta}
-parameters.assemblyReportUrl={'file:' if assembly_report else ''}{assembly_report}
-parameters.projects={projects}
-parameters.outputFolder={output_folder}
-'''
+        return self._format(
+            self._common_properties(assembly_accession=source_assembly, read_preference='secondaryPreferred',
+                                    chunk_size=1000),
+            {
+                'spring.batch.job.names': 'EXPORT_SUBMITTED_VARIANTS_JOB',
+                'parameters.taxonomy': taxonomy,
+                'parameters.fasta': fasta,
+                'parameters.assemblyReportUrl': self._format_str('file:{0}', assembly_report),
+                'parameters.projects': projects,
+                'parameters.outputFolder': output_folder
+            })
 
     def get_remapping_ingestion_properties(self, *, source_assembly='', target_assembly='', vcf='', load_to='',
                                            remapping_version=1.0):
         """Properties for remapping ingestion pipeline."""
-        return self._common_properties(target_assembly) + f'''
-spring.batch.job.names=INGEST_REMAPPED_VARIANTS_FROM_VCF_JOB
-parameters.vcf={vcf}
-parameters.remappedFrom={source_assembly}
-parameters.loadTo={load_to}
-parameters.remappingVersion={remapping_version}
-'''
-
-    def get_clustering_properties(self, *, instance,
-                                  job_name='', target_assembly='', rs_report_path='', projects='',
-                                  project_accession='', vcf='', source_assembly=''):
+        return self._format(
+            self._common_properties(assembly_accession=target_assembly, read_preference='secondaryPreferred',
```

Comment:
Suggest defaulting to 'primary' unless we have a reason not to in which case we can override it in the callsite...
```suggestion
            self._common_properties(assembly_accession=target_assembly, read_preference='primary',
```